plugins/header_rewrite/ipranges_helper.cc

```diff
@@ -0,0 +1,63 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+#include "ipranges_helper.h"
+
+#include <string_view>
+#include <vector>
+
+static std::vector<std::string_view>
+splitter(std::string_view input, char delim)
+{
+  std::vector<std::string_view> output;
+  size_t first = 0;
+
+  while (first < input.size()) {
```

Comment:
You need to use `TextView` here. This is precisely what it's designed for, instead of copy/pasting the same parsing logic over and over. E.g.
```
TextView src { input };
while (src) {
  auto token = input.take_prefix_at(delim);
  output.emplace_back(token);
}

Reply:
Changes to use ts::TextView, thanks for the suggestion.